src/Aspire.Hosting/ApplicationModel/ResourceRelationshipAnnotation.cs

```diff
@@ -0,0 +1,17 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Aspire.Hosting.ApplicationModel;
+
+#pragma warning disable RS0016 // Add public types and members to the declared API
+#pragma warning disable CS1591 // Missing XML comment for publicly visible type or member
+public class ResourceRelationshipAnnotation(IResource resource, string type, string? description) : IResourceAnnotation
```

Comment:
I'm not sure how I feel about the stringy `type`. I'm wondering whether it would be better to have a `ResourceRelationshipType` which can be inherited. It could then expose a way of getting a simple string value, but also provide a way of providing extra details around the relationship.